src/components/CheckEditor/FormComponents/HttpCheckAuthentication.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { MutableRefObject, useCallback, useRef, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { GrafanaTheme2 } from '@grafana/data';
+import { RadioButtonGroup, useStyles2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import { BasicAuth, CheckFormValuesHttp } from 'types';
+
+import { HttpCheckBasicAuthorization } from './HttpCheckBasicAuthorization';
+import { HttpCheckBearerToken } from './HttpCheckBearerToken';
+
+type HttpAuthType = 'none' | 'bearer' | 'basic';
+
+export const HttpCheckAuthentication = () => {
+  const styles = useStyles2(getStyles);
+
+  const { getValues, setValue } = useFormContext<CheckFormValuesHttp>();
+
+  const isBasicAuth = Boolean(
+    getValues(`settings.http.basicAuth.username`) || Boolean(getValues(`settings.http.basicAuth.password`))
+  );
+  const isBearerAuth = Boolean(getValues(`settings.http.bearerToken`));
+
+  const [authType, setAuthType] = useState<HttpAuthType>(isBasicAuth ? 'basic' : isBearerAuth ? 'bearer' : 'none');
+
+  const bearerTokenRef: MutableRefObject<string | undefined> = useRef(undefined);
+  const basicAuthRef: MutableRefObject<BasicAuth | undefined> = useRef(undefined);
```

Comment:
This is super nitty but just highlighting if you weren't aware that useRef can accept a generic 😄 

```suggestion
  const bearerTokenRef = useRef<string | undefined>(undefined);
  const basicAuthRef = useRef<BasicAuth | undefined>(undefined);
```